csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -15,442 +16,279 @@ namespace nvfuser::preseg_passes {
-  bool propagateAllocationOrder(TensorView* producer, TensorView* consumer) {
-    return propagateAllocationOrder(
-        producer, consumer, consumer->getMaybeRFactorDomain());
-  }
-
-  // Returns the candidate operand that dominates the allocation order.
-  //
-  // It scans through each candidate to find the first one that:
-  //   1. is a TensorView
-  //   2. has the most non_broadcast IterDomains
-  //
-  // The function returns a nullptr when it encounters a TensorView that does
-  // not have an entry in alloc_order_map_, since this means we failed to
-  // propagate memory format for an entry, we do NOT want to aggressively insert
-  // output memory format.
-  //
-  // The function is used to resolve allocation order propagation for operator
-  // with multiple operands. The operand with the most number of
-  // non-broadcast IterDomain will be dominating the output allocation order.
-  // The motivation behind it to avoid breaking allocation order propagation
-  // from operands produced by broadcast. e.g. When a binary operator could take
-  // in a channels_last 4d tensor and an unsqueezed bias vector. We'll want to
-  // propagate the channels_last allocation order to output.
-  //
-  // Pre-condition: `candidates` must be the input operands of the same Expr.
-  TensorView* resolveAllocationOrder(const std::vector<Val*>& candidates);
-
-  // alloc_order_map_ records the allocation order of each TensorView.
-  // Since it only handles permutation from a rfactor domain to allocation
-  // domain, it can be interpreted as:
-  //
-  // e.g. TV0 rfactor domain [i0, i1, i2]
-  //            alloc domain [i0, i2, i1]
-  //        allocation order   0,  2,  1
-  std::unordered_map<const TensorView*, AllocationOrder>& alloc_order_map_;
-};
-
-TensorView* AllocationOrderInferencer::resolveAllocationOrder(
-    const std::vector<Val*>& candidates) {
-  TensorView* src = nullptr;
-  size_t non_bc_high_water_mark = 0;
-
-  // helper utils to count the number of non broadcast / non reduction
-  // iterdomain
-  auto countLoopIterDomains = [](const TensorView* tv) -> size_t {
-    return std::count_if(
-        tv->getMaybeRFactorDomain().begin(),
-        tv->getMaybeRFactorDomain().end(),
-        [&](auto ptr_id) {
-          return !ptr_id->isBroadcast() && !ptr_id->isReduction();
-        });
-  };
-
-  for (auto* val : candidates) {
-    auto* tv = dynamic_cast<TensorView*>(val);
-    // skip non TensorView entry
-    if (tv == nullptr) {
+  // removing mapped ids and reduction ids to create unmapped_ids_vec.
+  // This means for the rest of ids in target_rfactor_domain that's not in
+  // mapped_id_set, they are either 1. a reduction domain, or; 2. in
+  // [unmapped_ids_vec.begin(), unmapped_ids_vec_end) This ensures that
+  // sharp-edges 1's loop would reconstruct a permutation of the
+  // target_rfactor_domain, hence a valid allocation domain for target.
+  std::vector<IterDomain*> unmapped_ids_vec = target_rfactor_domain;
```

Comment:
```suggestion
  std::vector<IterDomain*> unmapped_ids = target_rfactor_domain;
```
since you don't have a `_set` version.